tests/integration/publishergateway/test_write.py

```diff
@@ -0,0 +1,95 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Single-endpoint write tests (likely with a read query after)."""
+
+
+import contextlib
+import time
+
+import pytest
+from craft_store import errors, publishergateway
+
+from tests.integration.conftest import needs_charmhub_credentials
+
+
+@pytest.mark.slow
+@needs_charmhub_credentials()
+@pytest.mark.parametrize("version_pattern", [None, r"\d+"])
+@pytest.mark.parametrize("percentages", [None, 50])
+def test_create_tracks(
+    publisher_gateway: publishergateway.PublisherGateway,
+    charmhub_charm_name: str,
+    version_pattern,
+    percentages,
+):
+    track_name = str(time.time_ns())
+
+    tracks_created = publisher_gateway.create_tracks(
+        charmhub_charm_name,
+        {
+            "name": track_name,
+            "version-pattern": version_pattern,
+            "automatic-phasing-percentage": percentages,
+        },
+    )
+    assert tracks_created == 1
+
+    metadata = publisher_gateway.get_package_metadata(charmhub_charm_name)
+    if "tracks" not in metadata or not metadata["tracks"]:
+        raise ValueError("No tracks returned from the store")
```

Comment:
That's actually the reason why I prefer using `pydantic` models over `TypedDict` for known responses as you get validation for free and can access items using `.tracks` without worrying that it may not be there